### PR TITLE
Fixed read message of subscriber

### PR DESCRIPTION
**Pull request by t-asaka**
- Fixed timeout setting for reading messages
  - Set timeout only when shutdown subscriber and publisher is shutdown (for idle timed out)
- Fixed to close tcp conn when exiting `startRemotePublisherConn`

### CODE REVIEW

File: ros/subscriber.go [hunk 1]
```diff
@@ -192,51 +208,50 @@ func startRemotePublisherConn(logger Logger,
...
+				return
+			}
+			if err == io.EOF {
+				return
 			}
+			logger.Errorf("Failed to read a message size. err: %+v", err)
+			disconnectedChan <- pubUri
+			return
+		}
+		logger.Debugf("  %d", msgSize)
```
Comment (kt3k) [inline ros/subscriber.go]: I guess something like `"msgSize=%d"` would be more useful

**Review by kt3k — APPROVED** [review]

**Comment by t-asaka** [comment]
@at-wat Please take a look

File: ros/subscriber.go [hunk 2]
```diff
@@ -192,51 +208,50 @@ func startRemotePublisherConn(logger Logger,
 	}
 
 	// 3. Start reading messages
-	readingSize := true
-	var msgSize uint32 = 0
+	var quit bool
+	go func() {
+		<-quitChan
+		// set idle timeout
+		conn.SetDeadline(time.Now().Add(1000 * time.Millisecond))
```
Comment (at-wat) [inline ros/subscriber.go]:
```suggestion
		conn.SetDeadline(time.Now().Add(time.Second))
```

**Review by at-wat — APPROVED** [review]
LGTM
Sorry for the delay :bowing_man:

**Pull request merged**